### DISCUSSION TITLE
Vanilla or OOP ?

### DISCUSSION THREAD

**Discussion by ghost** [HuubMons/HuMo-genealogy#24] [Q&A]
The application was intended for procedural code but (poorly constructed) classes were implemented. Should we go in the direction of OOP or stay in procedural code?

**Comment by HuubMons** [comment]
I don't mind to change the code. And I do know the code can be improved.

My only tool for the last years is Notepad++. In the past I allready tried to find tools or frameworks (if possible free of cost) to improve the code. But It's not easy to find tools that will produce code that works for several users who install HuMo-genealogy at all kinds of exotic servers themself...

At this moment I'm trying to improve the results of the W3 validator (improving HTML, and solving problems in output).
And I'm busy changing the family tree editor.

And today I added a development branch at Github, and needed some time to read manuals about copy files to a second branch at Github.

I'm not a professional coder, and have a full time job. So sometimes it takes some time to update the code and learning new techniques...

**Reply by ghost** [reply]
Personally, I use vscode with the necessary plugins for php, html, css, js.... You have other IDEs like phpstorm but which are paid.

For testing tools there is a phpunit, Codeception and others but humogen is not currently testable. 
We need to refactor the code first by separating by features (view, model, controller).

But we have to choose between procedural and POO first.
And maybe rethink the organization/structure of the application.

And we will take the necessary time according to our possibilities.
